java/src/main/java/ai/rapids/cudf/ColumnView.java

```diff
@@ -1392,42 +1392,91 @@ public final ColumnVector rollingWindow(RollingAggregation op, WindowOptions opt
    * This is just a convenience method for an inclusive scan with a SUM aggregation.
    */
   public final ColumnVector prefixSum() {
-    return scan(Aggregation.sum());
+    return scan(this, Aggregation.sum());
   }
 
   /**
    * Computes a scan for a column. This is very similar to a running window on the column.
-   * @param aggregation the aggregation to perform
+   * @param input native id of a ColumnVector or ColumnView, a 0 (null column) is ok for
+   *              rank and dense rank aggregations.
+   * @param aggregation the aggregation to perform.
    * @param scanType should the scan be inclusive, include the current row, or exclusive.
    * @param nullPolicy how should nulls be treated. Note that some aggregations also include a
    *                   null policy too. Currently none of those aggregations are supported so
    *                   it is undefined how they would interact with each other.
    */
-  public final ColumnVector scan(Aggregation aggregation, ScanType scanType, NullPolicy nullPolicy) {
+  private static final ColumnVector scan(long input, Aggregation aggregation,
+      ScanType scanType, NullPolicy nullPolicy) {
     long nativeId = aggregation.createNativeInstance();
     try {
-      return new ColumnVector(scan(getNativeView(), nativeId,
+      return new ColumnVector(scan(input, nativeId,
           scanType.isInclusive, nullPolicy.includeNulls));
     } finally {
       Aggregation.close(nativeId);
     }
   }
 
+  /**
+   * Computes a scan for a column. This is very similar to a running window on the column.
+   * @param input input column that the scan operates on, a null value is ok for
+   *              rank and dense rank aggregations.
+   * @param aggregation the aggregation to perform.
+   * @param scanType should the scan be inclusive, include the current row, or exclusive.
+   *                 Note that rank and dense rank aggregates do not support exclusive scans.
+   * @param nullPolicy how should nulls be treated. Note that some aggregations also include a
+   *                   null policy too. Currently none of those aggregations are supported so
+   *                   it is undefined how they would interact with each other. Also note that
+   *                   rank and dense rank aggregates ignore the null policy.
+   */
+  public static final ColumnVector scan(ColumnVector input, Aggregation aggregation,
+      ScanType scanType, NullPolicy nullPolicy) {
+    if (input == null) return scan(0, aggregation, scanType, nullPolicy);
+    return scan(input.getNativeView(), aggregation, scanType, nullPolicy);
+  }
+
+  public static final ColumnVector scan(ColumnView input, Aggregation aggregation,
```

Comment:
This looks like it is duplicate code.

Reply:
Sorry I got it a bit wrong. In the java API all ColumnVectors are also ColumnViews, so you only need to keep the ColumnView versions.